packages/dashboard/src/components/sidePanel/sections/inputSettingsSection/index.spec.tsx

```diff
@@ -0,0 +1,120 @@
+});
+
+it('renders with options from state', () => {
+  const { container } = render(<TestComponent />);
+  const inputSettings = createWrapper(container);
+  const options = inputSettings.findTokenGroup('[data-test-id="input-widget-token-list"]');
+
+  MOCK_INPUT_WIDGET.options.forEach(({ label }, i) => {
+    expect(
+      options
+        ?.findToken(i + 1)
+        ?.findLabel()
+        .getElement().innerHTML
+    ).toBe(label);
+  });
+});
+
+it('can add option', () => {
+  const { container } = render(<TestComponent />);
+  const inputSettings = createWrapper(container);
+  const addOptionButton = inputSettings.findButton('[data-test-id="input-widget-add-option-btn"]');
+  const optionInput = inputSettings.findInput('[data-test-id="input-widget-option-input"]');
+  const options = inputSettings.findTokenGroup('[data-test-id="input-widget-token-list"]');
+  const newOption = 'lorem ipsum';
+
+  expect(addOptionButton?.isDisabled()).toBeTruthy();
+  expect(options?.findTokens().length).toBe(3);
+
+  optionInput?.setInputValue(newOption);
+
+  expect(addOptionButton?.isDisabled()).toBeFalsy();
+  addOptionButton?.click();
+
+  expect(options?.findTokens().length).toBe(4);
+  expect(options?.findToken(4)?.findLabel().getElement().innerHTML).toBe(newOption);
+});
+
+it('can remove option', () => {
+  const { container } = render(<TestComponent />);
+  const inputSettings = createWrapper(container);
+  const addOptionButton = inputSettings.findButton('[data-test-id="input-widget-add-option-btn"]');
+  const options = inputSettings.findTokenGroup('[data-test-id="input-widget-token-list"]');
+
+  expect(addOptionButton?.isDisabled()).toBeTruthy();
+  expect(options?.findTokens().length).toBe(3);
+
+  options?.findToken(2)?.findDismiss().click();
+
+  expect(options?.findTokens().length).toBe(2);
+  expect(options?.findToken(1)?.findLabel().getElement().innerHTML).toBe(widget.options[0].label);
+  expect(options?.findToken(2)?.findLabel().getElement().innerHTML).toBe(widget.options[2].label);
+});
+
+it('correctly renders translations', () => {
+  const optionPlaceholder = 'lorem ipsum';
+  const addOptionLabel = 'lorem ipsum 2';
+  const props = {
+    messageOverride: {
+      ...DefaultDashboardMessages,
+      sidePanel: {
+        ...DefaultDashboardMessages.sidePanel,
+        inputSettings: {
+          ...DefaultDashboardMessages.sidePanel.inputSettings,
+          optionPlaceholder,
+          addOptionLabel,
+        },
+      },
+    },
+  };
+  const { container } = render(<TestComponent {...props} />);
+  const inputSettings = createWrapper(container);
+  const addOptionButton = inputSettings.findButton('[data-test-id="input-widget-add-option-btn"]');
+  const optionInput = inputSettings.findInput('[data-test-id="input-widget-option-input"]');
+
+  expect(addOptionButton?.findTextRegion()?.getElement().innerHTML).toBe(addOptionLabel);
+  expect(optionInput?.findNativeInput().getElement().placeholder).toBe(optionPlaceholder);
```

Comment:
instead of searching down the DOM tree, we should be using best practices testing-library where possible. 

For example, there is a `getByPlaceholderText` API, as well as a way to say `get button that says 'lorem ipsum 2' instead of `findTextRegion().getElement().innerHTML.

Reply:
I am using the cloudescape testing lilbrary for this https://cloudscape.design/get-started/testing/introduction/